mlflow/langchain/utils/chat.py

```diff
@@ -345,3 +347,55 @@ def is_list_of_chat_messages(json_message: list[dict]):
             return request_json, False
     else:
         return request_json, False
+
+
+def parse_token_usage(
+    lc_generations: list[Generation],
+) -> Optional[dict[str, int]]:
+    """Parse the token usage from the LangChain generations."""
+    aggregated = defaultdict(int)
+    for generation in lc_generations:
+        if token_usage := _parse_token_usage_from_generation(generation):
+            for key in token_usage:
+                aggregated[key] += token_usage[key]
+
+    return dict(aggregated) if aggregated else None
+
+
+def _parse_token_usage_from_generation(generation: Generation) -> Optional[dict[str, int]]:
+    message = getattr(generation, "message", None)
+    if not message:
+        return None
+
+    metadata = (
+        message.usage_metadata
+        or message.response_metadata.get("usage")
+        or message.response_metadata.get("token_usage")
+    )
+    return _parse_token_counts(metadata) if metadata else None
+
+
+def _parse_token_counts(usage_metadata: dict[str, Any]) -> dict[str, int]:
+    """Standardize token usage metadata keys to MLflow's token usage keys."""
+    key_mapping = {
+        # OpenAI
+        "prompt_tokens": TokenUsageKey.INPUT_TOKENS,
+        "completion_tokens": TokenUsageKey.OUTPUT_TOKENS,
+        "total_tokens": TokenUsageKey.TOTAL_TOKENS,
+        # OpenAI Streaming, Anthropic, etc.
+        "input_tokens": TokenUsageKey.INPUT_TOKENS,
+        "output_tokens": TokenUsageKey.OUTPUT_TOKENS,
+    }
+
+    usage = {}
+    for key, value in usage_metadata.items():
+        if key in key_mapping:
+            usage[key_mapping[key]] = value
+
+    # If the total tokens are not present, calculate it from the input and output tokens
+    if usage and usage.get(TokenUsageKey.TOTAL_TOKENS) is None:
+        input_tokens = usage.get(TokenUsageKey.INPUT_TOKENS, 0)
+        output_tokens = usage.get(TokenUsageKey.OUTPUT_TOKENS, 0)
+        usage[TokenUsageKey.TOTAL_TOKENS] = input_tokens + output_tokens
```

Comment:
```suggestion
        usage[TokenUsageKey.TOTAL_TOKENS] = usage.get(TokenUsageKey.INPUT_TOKENS, 0) + usage.get(TokenUsageKey.OUTPUT_TOKENS, 0)
``` 
We don't use those variables later and it's clear what it is :)